mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxExtensions.kt

```diff
@@ -54,6 +55,53 @@ inline fun <T, reified VM : BaseMvRxViewModel<S>, reified S : MvRxState> T.activ
         .apply { subscribe(this@activityViewModel, subscriber = { postInvalidate() }) }
 }
 
+/**
+ * Gets or creates a Fragment-scoped ViewModel scoped to this View. You will get the same instance every time for this Fragment, even
+ * through rotation, or other configuration changes.
+ *
+ * This will walk up the view hierarchy until it finds a View that is the root view for a Fragment.
+ *
+ * If the ViewModel has additional dependencies, implement [MvRxViewModelFactory] in its companion object.
+ * You will be given the initial state as well as a FragmentActivity with which you can access other dependencies to
+ * pass to the ViewModel's constructor.
+ *
+ * MvRx will also handle persistence across process restarts. Refer to [PersistState] for more info.
+ *
+ * Use [keyFactory] if you have multiple ViewModels of the same class in the same scope.
+ */
+inline fun <T, reified VM : BaseMvRxViewModel<S>, reified S : MvRxState> T.fragmentViewModel(
+    viewModelClass: KClass<VM> = VM::class,
+    crossinline keyFactory: () -> String = { viewModelClass.java.name }
+) where T : View, T : StatefulView = viewLifecycleAwareLazy(this) { activity, fragment ->
+    MvRxViewModelProvider.get(viewModelClass.java, S::class.java, FragmentViewModelContext(activity, fragment._fragmentArgsProvider(), fragment), keyFactory())
+        .apply { subscribe(fragment.viewLifecycleOwner, subscriber = { onInvalidate() }) }
+}
+
+/**
+ * [fragmentViewModel] except scoped to the current Activity. Use this to share state between different Fragments.
```

Comment:
Some people use view architectures directly without fragments - it would be nice to make activity/existing view model not related to fragments (ie remove the `share state between different Fragments.` wording)